chat/public/js/src/chat.js

```diff
@@ -361,9 +361,8 @@ function ChatXBlock(runtime, element, init_data) {
      * pause: delays execution with a timeout
      */
     var pause = function(timeout) {
-        return new Promise(function(resolve, reject) {
-            setTimeout(resolve, timeout);
-        });
+        var promise = $.Deferred();
+        setTimeout(promise.resolve, timeout);
```

Comment:
@carlio It looks like the `return promise;` disappeared after rebase? I think that's what's causing the test to fail - if it doesn't return the promise then the functions using `pause` will execute immediately and not wait for that `setTimeout`.